packages/ember-glimmer/tests/integration/components/attrs-lookup-test.js

```diff
@@ -57,32 +58,44 @@ moduleFor('Components test: attrs lookup', class extends RenderingTest {
 
   ['@htmlbars should be able to modify a provided attr into local state #11571 / #11559'](assert) {
     let instance;
-
     let FooBarComponent = Component.extend({
       init() {
         this._super(...arguments);
         instance = this;
       },
 
       didReceiveAttrs() {
-        this.set('first', this.attrs.first.toUpperCase());
+        this.set('first', this.getAttr('first').toUpperCase());
       }
     });
     this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template: '{{first}}' });
 
-    this.render(`{{foo-bar first="first attr"}}`);
+    this.render(`{{foo-bar first=firstAttr}}`, {
+      firstAttr: 'first attr'
+    });
 
-    assert.equal(instance.get('first'), 'FIRST ATTR');
+    assert.equal(instance.get('first'), 'FIRST ATTR', 'component lookup uses local state');
     this.assertText('FIRST ATTR');
 
     this.runTask(() => this.rerender());
 
-    assert.equal(instance.get('first'), 'FIRST ATTR');
+    assert.equal(instance.get('first'), 'FIRST ATTR', 'component lookup uses local state during rerender');
+    this.assertText('FIRST ATTR');
+
+    this.runTask(() => set(this.context, 'firstAttr', 'second attr'));
+
+    assert.equal(instance.get('first'), 'SECOND ATTR', 'component lookup uses modified local state');
```

Comment:
I think there is a legit bug here. I pulled this locally and placed the a currly on the outside of the component to make sure things were sane. It appears `didReceiveAttrs` is not actually being re-triggered.
<img width="608" alt="screen shot 2016-04-07 at 10 49 32 pm" src="https://cloud.githubusercontent.com/assets/183799/14375142/15e6dfc2-fd13-11e5-86b7-364ee273ddf3.png">

I then decided to see if this was a shadowing issue with the `attr` and local property being the same. The componet code changed to setting a local property called `local` and using it in the template like `{{local}}`.

```
let FooBarComponent = Component.extend({
  init() {
     this._super(...arguments);
     instance = this;
     this.local = null;
  },
  didReceiveAttrs() {
    this.set('local', this.getAttr('first').toUpperCase());
  }
});
```
This resulted in an even more bizarre output.
<img width="615" alt="screen shot 2016-04-07 at 11 14 56 pm" src="https://cloud.githubusercontent.com/assets/183799/14375499/98b1338c-fd16-11e5-96f8-784a28acc64c.png">

So it started updating but then the attr was not uppercase. Adding @rwjblue since he may have more context since he added the original tests around this.